src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs

```diff
@@ -116,6 +118,7 @@ protected override void ProcessRecordCore()
                 await _lockService.ExecuteNuGetOperationAsync(() =>
                 {
                     SubscribeToProgressEvents();
+                    ValidateArgumentsAreSupported();
```

Comment:
Should we differentiate between update for all projects vs individual selected project? And then according to selected project, takes the call to show this warning?

Reply:
That's already done in an non-intuitive way. 
Had me confused initially as well. 
https://github.com/NuGet/NuGet.Client/blob/bf99156406e4e702bd3e11ad1ffe79f7571941d1/src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs#L84-L91